postgres-appliance/launch.sh

```diff
@@ -46,6 +46,15 @@ elif python3 /scripts/configure_spilo.py all; then
     else
         $CMD &
     fi
+else ## /scripts/configure_spilo failed.  Try one more time after a brief pause.
+    echo "/launch.sh: /scripts/configure_spilo.py failed.  Retrying after a brief delay..."
+    sleep 2
+    CMD="/scripts/patroni_wait.sh -t 3600 -- envdir $WALE_ENV_DIR /scripts/postgres_backup.sh $PGDATA"
+    if [ "$(id -u)" = "0" ]; then
+        su postgres -c "PATH=$PATH $CMD" &
+    else
+        $CMD &
+    fi
```

Comment:
I think you don't understand what you are doing.

Non-zere exit code in configure_spilo.py doesn't always mean that it failed:
https://github.com/zalando/spilo/blob/ab4ddee8eae561373eaa272b5a82d12b6621f8fc/postgres-appliance/scripts/configure_spilo.py#L1110-L1111

If retry should happen, you better do it in the configure_spilo.py itself.

But, if we think a bit harder and analyze the problem, it becomes clear that we are dealing with a classical race condition. Normally such kinds of problems are solved with the help of init containers and spilo shouldn't even start until the full initialization is completed.

Reply:
I did miss the fact that the exit code is being abused in that manner.  That does indeed complicate matters.

I'll investigate the init container approach you've suggested.